app.py

```diff
@@ -201,60 +202,15 @@ def cb():
             }
 
 
-schema_name = "authd4"
-directory_name = "authd4"
-schema_version = "1"
-directory_arn = None
-
-for directory in ad.list_directories()["Directories"]:
-    if directory["Name"] == directory_name:
-        directory_arn = directory["DirectoryArn"]
-        break
-
-schema_arn = "{}/schema/{}/{}".format(directory_arn, schema_name, schema_version)
-
-
-def get_object_attribute_list(facet="User", **kwargs):
-    return [dict(Key=dict(SchemaArn=schema_arn, FacetName=facet, Name=k), Value=dict(StringValue=v))
-            for k, v in kwargs.items()]
-
-
-def provision_user(email):
-    username = quote(email)
-    default_group = service_config["Services"][0]["DefaultProvisioningGroups"][0]
-    policy = service_config["Services"][0]["DefaultGroupPolicies"][default_group]
-    for statement in policy["Statement"]:
-        statement["Resource"] = statement["Resource"].format(user_id=email)
-    print("POLICY:")
-    print(policy)
-    try:
-        res = ad.create_object(DirectoryArn=directory_arn,
-                               SchemaFacets=[dict(SchemaArn=schema_arn, FacetName="User")],
-                               ObjectAttributeList=get_object_attribute_list(username=username, email=email),
-                               ParentReference=dict(Selector="/"),
-                               LinkName=username)
-        print(res["ObjectIdentifier"])
-    except ad.exceptions.LinkNameAlreadyInUseException as e:
-        print(e)
-    res = ad.create_object(
-        DirectoryArn=directory_arn,
-        SchemaFacets=[dict(SchemaArn=schema_arn, FacetName="IAMPolicy")],
-        ObjectAttributeList=[
-            dict(Key=dict(SchemaArn=schema_arn, FacetName="IAMPolicy", Name="policy_type"),
-                 Value=dict(StringValue="IAMPolicy")),
-            dict(Key=dict(SchemaArn=schema_arn, FacetName="IAMPolicy", Name="policy_document"),
-                 Value=dict(BinaryValue=json.dumps(policy).encode()))
-        ]
-    )
-    policy_id = res["ObjectIdentifier"]
-    for pid in ad.list_object_policies(DirectoryArn=directory_arn,
-                                       ObjectReference=dict(Selector="/" + username))["AttachedPolicyIds"]:
-        ad.detach_policy(DirectoryArn=directory_arn,
-                         PolicyReference=dict(Selector="$" + pid),
-                         ObjectReference=dict(Selector="/" + username))
-    ad.attach_policy(DirectoryArn=directory_arn,
-                     PolicyReference=dict(Selector="$" + policy_id),
-                     ObjectReference=dict(Selector="/" + username))
+directory_name = os.getenv("FUSILLADE_DIR", f"hca_fusillade_{os.environ['FUS_DEPLOYMENT_STAGE']}")
+try:
+    directory = CloudDirectory.from_name(directory_name)
+except FusilladeException:
+    from fusillade.clouddirectory import publish_schema, create_directory
+    schema_name = f"hca_fusillade_{os.environ['FUS_DEPLOYMENT_STAGE']}"
```

Comment:
`schema_name` is the default for `directory_name` and seems like it should be declared before it since it's used in 2 spots, as in:
```
schema_name = f"hca_fusillade_{os.environ['FUS_DEPLOYMENT_STAGE']}"
directory_name = os.getenv("FUSILLADE_DIR", schema_name)
try:
    schema_version = "1"
    directory = CloudDirectory.from_name(directory_name)
    directory_arn = None
except FusilladeException:
    from fusillade.clouddirectory import publish_schema, create_directory
    schema_version = "0.1"
    schema_arn = publish_schema(schema_name, schema_version)
    directory = create_directory(directory_name, schema_arn)
```

Reply:
The schema and directory are different resources. I will change the name of schema_name